src/routes/categories/containers/__tests__/LanguageFailure.spec.js

```diff
@@ -0,0 +1,147 @@
+                       t={key => key} />
+    ).instance()
+
+    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(1)
+    expect(mockSetLanguageChangeUrls).toBeCalledWith(categoriesPage.mapLanguageToUrl, languages)
+  })
+
+  test('should map language to url', () => {
+    const mapLanguageToUrl = shallow(
+      <LanguageFailure locations={locations}
+                       languages={languages}
+                       location={location}
+                       language={language}
+                       setLanguageChangeUrls={() => {}}
+                       t={key => key} />
+    ).instance().mapLanguageToUrl
+
+    expect(mapLanguageToUrl(language)).toBe(`/${location}/${language}`)
+  })
+
+  describe('connect', () => {
+    const languagesEndpoint = new EndpointBuilder('languages')
+      .withUrl('https://weird-endpoint/api.json')
+      .withMapper(json => json)
+      .withResponseOverride(languages)
+      .build()
+
+    const locationsEndpoint = new EndpointBuilder('locations')
+      .withUrl('https://weird-endpoint/api.json')
+      .withMapper(json => json)
+      .withResponseOverride(locations)
+      .build()
+
+    const pathname = '/augsburg/tu'
+    const id = '1234'
+
+    test('should map state to props', () => {
+      const store = createReduxStore(createHistory, {
+        router: {
+          params: {location: location, language: language},
+          pathname: pathname,
+          query: {id: id}
+        },
+        languageChangeUrls: {}
+      })
+
+      const languageFailure = mount(
+        <Provider store={store}>
+          <EndpointProvider endpoints={[locationsEndpoint, languagesEndpoint]}>
+            <ConnectedLanguageFailure />
+          </EndpointProvider>
+        </Provider>
+      ).find(LanguageFailure)
+
+      expect(languageFailure.props()).toEqual({
+        location: location,
+        setLanguageChangeUrls: expect.any(Function),
+        t: expect.any(Function),
+        locations: locations,
+        languages: languages
+      })
+    })
+
+    test('should map dispatch to props', () => {
+      const store = createReduxStore(createHistory, {
+        router: {
+          params: {location: location, language: language},
+          pathname: pathname,
+          query: {id: id}
+        },
+        languageChangeUrls: {}
+      })
+
+      const mapLanguageToUrl = (language) => `/${language}`
+
+      const languageChangeUrls = {
+        en: '/en',
+        ar: '/ar',
+        de: '/de'
+      }
+
+      expect(store.getState().languageChangeUrls).not.toEqual(languageChangeUrls)
+
+      const languageFailure = mount(
+        <Provider store={store}>
+          <EndpointProvider endpoints={[locationsEndpoint, languagesEndpoint]}>
+            <ConnectedLanguageFailure />
+          </EndpointProvider>
+        </Provider>
+      ).find(LanguageFailure)
+
+      languageFailure.props().setLanguageChangeUrls(mapLanguageToUrl, languages)
```

Comment:
I think setLanguageChangeUrls should already get called by mount()

Reply:
Right